src/seedu/addressbook/data/person/Address.java

```diff
@@ -20,13 +20,70 @@
      *
      * @throws IllegalValueException if given address string is invalid.
      */
+    private class Block {
```

Comment:
Some things to think about:
Firstly, the argument name (value) may not be intuitive.
Second, think about the benefits of having classes in their own files VS nested classes (like what you are doing now)
Finally, think about whether there are any benefit in having these classes private. What about making them public?